app/models/notification.rb

```diff
@@ -0,0 +1,34 @@
+class Notification < ActiveRecord::Base
+  acts_as_readable on: :updated_at
+  belongs_to :user
+  belongs_to :course
+
+  scope :center_popup, -> { where(type: 'CenterPopup') }
+  scope :right_side_popup, -> { where(type: 'RightSidePopup') }
+  scope :email, -> { where(type: 'EmailNotification') }
+
+  def self.types
+    %w(CenterPopup RightSidePopup EmailNotification)
+  end
+
+  def self.create(user, course, options = {})
```

Comment:
Also I don't understand how to use your function, what are the available options ?   Can you document this or put this in the PR description. 

The most important thing is to tell others all your API functions and how to use them, because your are developing a library for notification.